data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwarderAppConfig.java

```diff
@@ -96,15 +100,28 @@ public ResponseHandler responseHandler(@Qualifier("peerForwarderMetrics") final
         return new ResponseHandler(pluginMetrics);
     }
 
+    @Bean
```

Comment:
You could create a single bean which introspects the `binary_coded` property and then returns the correct bean. This would help with my broader refactoring suggestions.